### DISCUSSION TITLE
classes/class_joint3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#179] [User-contributed notes]
# classes/class_joint3d

Inherits: Node3D< Node< Object Inherited By: ConeTwistJoint3D, Generic6DOFJoint3D, HingeJoint3D, PinJoint3D, SliderJoint3D Abstract base class for all 3D physics joints. Description: Abstract base ...

https://docs.godotengine.org/en/stable/classes/class_joint3d.html

**Comment by mishapsi** [comment]
Dynamically instancing joints:

This was a bit of a headscratcher so hopefully this saves someone time.  Before setting NodeA and NodeB, set the position of the joint first, otherwise NodeA and NodeB will be initialized with a joint at world 0,0,0 despite setting the joint's position to something else.
